### DISCUSSION TITLE
Changing sqlite version as required

### DISCUSSION THREAD

**Discussion by tshrwbl** [pawelsalawa/sqlitestudio#3897] [Q&A]
I wanted to test code with different versions of SQLite. Is there any way to do it?

**Comment by pawelsalawa** [comment]
You can simply replace sqlite library binary file. What operating system do you have?

**Reply by tshrwbl** [reply]
Windows 10.  What's the location of SQLite binary file?

**Reply by pawelsalawa** [reply]
Same directory as `SQLiteStudio.exe` (very likely `C:\Program Files\SQLiteStudio\`). Look for `sqlite3.dll`.

**Reply by tshrwbl** [reply]
Ok got it, Thanks. Can I go higher than default version 3.24? getting some issue with 3.30+

**Reply by pawelsalawa** [reply]
You can use whatever version you like, just be aware that GUI components, SQL parsers, etc - are not adjusted to support some of new features, therefore you may see some warnings or errors here and there. Nevertheless, the SQL Editor window should always be able to execute SQL query, no matter what new features you use. It will underline new syntax as erronous, but you still can execute it, as long as the sqlite3.dll is new and supports it.

**Reply by vanderdoes** [reply]
Hi, if I try to copy sqlite3.dll from a newer version to the SQLiteStudio directory and then start SQLiteStudio it fails.  Is there another place that a config needs to be updated for SQLiteStudio to start properly?  --Thanks Jeff

**Reply by pawelsalawa** [reply]
No other place needs to be changed. Apparently there are some changes in 3.30+ that are incompatible enough to make it fail. Unfortunately.

**Reply by vanderdoes** [reply]
Thanks for response!  Unfortunately 3.28 also fails saying The application was unable to start correctly (0xc000007b).  Are there thoughts to release a newer version that will support later versions of sqlite?

**Reply by pawelsalawa** [reply]
I really struggle with time to finalize 3.2.2 :/ I'd like to finish it as soon as I can. I don't know when it will be though.

**Comment by iheathers** [comment]
How do i do it for mac?

**Reply by pawelsalawa** [reply]
Download the binary from sqlite.org, unzip it. Then copy & overwrite `libsqlite3.dylib` (or somewhat similar file) in the `SQLiteStudio.app/Contents/Framework/` (or maybe a bit different folder, but it's rather easy to find, look around).
